pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -376,33 +369,37 @@ func reduceQuery(query string, reduceTo v3.ReduceToOperator, aggregateOperator v
 	return query, nil
 }
 
-func addLimitToQuery(query string, limit uint64, panelType v3.PanelType) string {
-	if limit == 0 {
-		limit = 100
-	}
-	if panelType == v3.PanelTypeList {
-		return fmt.Sprintf("%s LIMIT %d", query, limit)
-	}
-	return query
+func addLimitToQuery(query string, limit uint64) string {
+	return fmt.Sprintf("%s LIMIT %d", query, limit)
 }
 
 func addOffsetToQuery(query string, offset uint64) string {
 	return fmt.Sprintf("%s OFFSET %d", query, offset)
 }
 
-func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery, fields map[string]v3.AttributeKey) (string, error) {
-	query, err := buildLogsQuery(start, end, mq.StepInterval, mq, fields)
+func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery) (string, error) {
+	query, err := buildLogsQuery(panelType, start, end, mq.StepInterval, mq)
 	if err != nil {
 		return "", err
 	}
 	if panelType == v3.PanelTypeValue {
 		query, err = reduceQuery(query, mq.ReduceTo, mq.AggregateOperator)
 	}
 
-	query = addLimitToQuery(query, mq.Limit, panelType)
-
-	if mq.Offset != 0 {
-		query = addOffsetToQuery(query, mq.Offset)
+	if panelType == v3.PanelTypeList {
+		if mq.Limit == 0 {
+			// 100 is the default max limit as well as page limit
+			mq.Limit = 100
+		}
+		if mq.PageSize > 0 {
+			if mq.Offset > mq.Limit {
+				return "", fmt.Errorf("max limit exceeded")
+			}
+			query = addLimitToQuery(query, mq.PageSize)
+			query = addOffsetToQuery(query, mq.Offset)
```

Comment:
Please help me with this. I want to browse several pages until I find what I want. The limit is not set. I set the page size to 50. After three pages, will this give the results or throw an error?

Reply:
Yeah, this case you mentioned will result in error after three pages. From my side I don't think that should be an issue.

What are your thoughts ?

Reply:
I believe that is unexpected from an end-user POV. When a limit is not applied, it shouldn't give an error when I go to the third/fourth page. Can you share why you want to raise `max limit exceeded`? Like what was the reasoning to introduce this error?

Reply:
In in table view(groupby), the UI will have some set default limit which the user can change. Now during pagination if the limit is hit then it should be stopped, thats is POV for adding this error.

Reply:
It sets the limit as null by default. It has to be changed by the user. Should you be setting the limit (to 100) when no limit is applied on the frontend? It is fair to throw an error if they _set_ the limit and request for data beyond that, but if they don't set any limit and ask for data, we should return data.

Reply:
Yeah don't see an issue with that let me add it.